crates/state/src/batch.rs

```diff
@@ -73,6 +60,19 @@ impl BatchCheckpoint {
     }
 }
 
+impl From<BatchCheckpoint> for ProofReceipt {
```

Comment:
I think it's better to have a different method that does this instead of `From`/`Into` implementation. Here's a [nice guideline on when to implement `From`](https://doc.rust-lang.org/std/convert/trait.From.html#when-to-implement-from)